includes/abilities-api/class-wp-abilities-registry.php

```diff
@@ -64,6 +64,16 @@ final class WP_Abilities_Registry {
 	 * } $args
 	 */
 	public function register( string $name, array $args ): ?WP_Ability {
+		/**
+		 * Filters the ability arguments before they are validated.
+		 *
+		 * @since n.e.x.t
+		 *
+		 * @param array<string,mixed> $args The arguments used to instantiate the ability.
+		 * @param string              $name The name of the ability, with its namespace.
+		 */
+		$args = apply_filters( 'wp_ability_args', $args, $name );
```

Comment:
Minor note, this could get moved after all checks against `$name` to ensure it doesn't get processed on abilities that have an incorrect name.

Reply:
No worries, it's why additional eyes are helpful to the implementation from a different perspective 👍🏻